### ISSUE TITLE
false positive of or_fun_call for enum

### ISSUE THREAD

**Issue by xxchan** [rust-lang/rust-clippy#9663]
### Summary

See Reproducer. Note that in my trials, this only happens when one variant is `std::io::Error`

### Lint Name

or_fun_call

### Reproducer

I tried this code:

```rust
pub enum E1 {
    A(usize),
    B(std::io::Error),
}

pub fn foo() -> Result<(), E1> {
    None.ok_or(E1::A(1))
}
```

I saw this happen: 

```
no warning
```

I expected to see this happen:
```
warning: use of `ok_or` followed by a function call
  --> src/tmp1/src/main.rs:61:10
   |
61 |     None.ok_or(E1::A(1))
   |          ^^^^^^^^^^^^^^^ help: try this: `ok_or_else(|| E1::A(1))`
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#or_fun_call
   = note: `#[warn(clippy::or_fun_call)]` on by default
```

### Version

```text
rustc 1.66.0-nightly (b8c35ca26 2022-10-15)
binary: rustc
commit-hash: b8c35ca26b191bb9a9ac669a4b3f4d3d52d97fb1
commit-date: 2022-10-15
host: aarch64-apple-darwin
release: 1.66.0-nightly
LLVM version: 15.0.2
```

**Comment by kraktus** [comment]
probably fixed already by https://github.com/rust-lang/rust-clippy/pull/9610

**Comment by J-ZhengLi** [comment]
Same as #9693, it appears to be already fixed, so I'm closing this, thank you for the report 👀
